admin-dev/themes/new-theme/js/pages/product/edit/index.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ProductMap from '../product-map';
+import ProductPartialUpdater from './product-partial-updater';
+
+const {$} = window;
+
+$(() => {
+  window.prestashop.component.initComponents(
+    [
+      'TranslatableField',
+      'TinyMCEEditor',
+      'TranslatableInput',
+    ],
+  );
+
+  const $productForm = $(ProductMap.productForm);
+
+  // Form has productId data means that we are in edtion/update mode
```

Comment:
Maybe we should not rely on this (I dont know what crazy stuff could happen) but rather send from backend a variable telling us the mode?

And also we could send more data like "combinations or not, virtual, pack..." from backend to frontend?

Reply:
This could be changed later if we need extra variables in which case we can rely on a different one
But in the end we will still need to add some extra `data-something` in the HTML so that it is parsed here, so `productId` seemed like a good start since indeed we don't know what will be needed later 🤷‍♂️  So we can improve it later